File: CodingTestPlatform/BOJ/2_Silver/4/1302/1302.py
# 베스트셀러
# https://www.acmicpc.net/problem/1302

import sys

input = sys.stdin.readline
N = int(input().rstrip())

book_map = {}
for _ in range(N):
    book = input().rstrip()
    if book in book_map:
        book_map[book] += 1
    else:
        book_map[book] = 1

rst = []
for key in book_map:
    rst.append((book_map[key], key))

rst.sort(key=lambda x: (-x[0], x[1]))
print(rst[0][1])

File: CodingTestPlatform/BOJ/2_Silver/4/1302/1302.js
/**
 * 베스트셀러
 * Silver4
 * https://www.acmicpc.net/problem/1302
 */

const [N, ...books] = require('fs').readFileSync('/dev/stdin').toString().trim().split("\n");

const hashMap = {};
let max = 0;

books.forEach((book) => {
    !hashMap[book] ? hashMap[book] = 1 : hashMap[book]++;
    max = Math.max(max, hashMap[book]);
});

const result = books.filter((book) => hashMap[book] === max).sort()[0];

console.log(result);